hypothesis-python/tests/cover/test_recursive.py

```diff
@@ -77,3 +79,29 @@ def test_can_exclude_branching_with_max_leaves(t):
 @given(st.recursive(st.none(), lambda x: st.one_of(x, x)))
 def test_issue_1502_regression(s):
     pass
+
+
+def test_thread_save_draw():
+    shared_strategy = st.recursive(st.integers(), lambda s: st.lists(s, max_size=3))
+
+    errors = []
+
+    @given(data=st.data())
+    def test(data):
+        try:
+            data.draw(shared_strategy)
+        except Exception as exc:
+            errors.append(exc)
+
+    threads = []
+
+    for _ in range(2):
```

Comment:
Is two threads enough to reliably find this error without the patch?  If in doubt, I'd prefer to make this >= 95% reliable and move it to `tests/nocover/` since it's likely to be slow and doesn't have unique coverage.

Reply:
Yep, 2 threads trigger the bug in 77/100 test runs. 4 threads do 998/1000. I moved the test to `nocover` & changed the number of threads to 4